snapcraft/plugins/nodejs.py

```diff
@@ -61,25 +64,23 @@ def schema(cls):
             'items': {
                 'type': 'string'
             },
-            'default': [],
+            'default': []
         }
         schema['properties']['node-engine'] = {
             'type': 'string'
         }
-
-        if 'required' in schema:
-            del schema['required']
+        # `node-engine` keyword is required, but `node-packages` is not.
+        schema['required'].append('node-engine')
+        schema['required'].remove('source')
+        schema['properties']['source']['default'] = '.'
```

Comment:
Can we leave this one for later? `default` is a contentious topic and would be better applied across the board